core/include/scipp/core/transform.h

```diff
@@ -524,25 +525,16 @@ template <bool dry_run> struct in_place {
               is_ValuesAndVariances_v<std::decay_t<decltype(arg)>>;
           constexpr bool args_var =
               (is_ValuesAndVariances_v<std::decay_t<decltype(args)>> || ...);
-          constexpr bool force_same = std::is_base_of_v<
-              transform_flags::expect_all_or_none_have_variance_t, Op>;
-          if constexpr (force_same &&
-                        check_all_or_none_variances(
-                            is_ValuesAndVariances_v<
-                                std::decay_t<decltype(arg)>>,
-                            is_ValuesAndVariances_v<
-                                std::decay_t<decltype(args)>>...)) {
-            (void)arg_var;
-            (void)args_var;
-            (void)in_var_if_out_var;
+          if constexpr (check_all_or_none_variances_v<
+                            Op, std::decay_t<decltype(arg)>,
+                            std::decay_t<decltype(args)>...>) {
```

Comment:
All the `decay_t` could be moved into the trait for cleaner code at the call sites.